### ISSUE TITLE
After loss.backward(),  why needs torch.autograd.backward(inputs, grads, retain_graph=retain_graph)

### ISSUE THREAD

**Issue by LarryLee-BD** [ratishsp/data2text-plan-py#11]
Hi Ratish, I have a problem here. After __compute_loss()_ and _loss.div(normalization).backward()_, why in function _shards_ , there is _torch.autograd.backward(inputs, grads, retain_graph=retain_graph)_  ? is it usual we use loss to do backward job? why use inputs backward again?

**Comment by ratishsp** [comment]
Hi @LarryLee-BD, this repo is based upon a fork of [OpenNMT-py](https://github.com/OpenNMT/OpenNMT-py). This thread https://github.com/OpenNMT/OpenNMT-py/issues/387 gives the details of the two ```backward()``` operations.
